LetSwift/Extensions/UIImageExtension.swift

```diff
@@ -29,4 +29,28 @@ extension UIImage {
         
         return UIGraphicsGetImageFromCurrentImageContext()
     }
+    
+    func tinted(with color: UIColor) -> UIImage {
+        guard let filter = CIFilter(name: "CIColorMonochrome") else { return self }
+        
+        filter.setValue(CIColor(color: color), forKey: kCIInputColorKey)
+        
+        return filtered(by: filter) ?? self
+    }
+    
+    private func filtered(by filter: CIFilter) -> UIImage? {
+        guard let ciImage = CIImage(image: self) else { return nil }
```

Comment:
To get the `CIImage` from `UIImage` usually you would have to first check if `UIImage` has associated `CIImage` with it (`image.ciImage`, wondering if this is the same as `CIImage(image:)` 🤔 ), or if it doesn't (which sometimes happen), you would then proceed to use `CGIImage` to get `CIImage` (`CIImage(cgImage: image.cgImage)`).

Reply:
https://stackoverflow.com/a/15651358 <- I think here is the enough good explanation

Reply:
@neegra Hm, didn't understand the answer fully, but after reading the docs it seems logical - these two properties (`cgImage`, `ciImage`) are only associated backing platforms of `UIImage`. Unfortunately I didn't found any info about `CIImage(image:)` initializer though - how is the `CIImage` created from `UIImage` or when will it return `nil`?

Honestly didn't hear about this initializer before, if this is equivalent of what I wrote above, that would be cool!

Reply:
I'll try to explain by code:

``` Swift
let image = UIImage()
let ciImage = CIImage(image: image) //result: nil

let image2 = UIImage()
image2.ciImage // result: nil
image2.cgImage // result: nil

let image3 = UIImage(ciImage: CIImage())
image3.ciImage // result: non-nil
```

Reply:
1) When you create UIImage with `CIImage` or `CGImage` it contains the source which was made of. Accordingly, when from `CIImage` then `cgImage` is `nil`,  and when from `CGImage` then `ciImage` is `nil`. 
2) If it's going about `CIImage` created from `UIImage` then it may be nil, when `UIImage` is malformed in any way, e.g. doesn't contain any data.

Reply:
But I cannot find anything what describes what is under hood the `CIImage.init?(image: UIImage)`

Reply:
@sunshinejr Apparently `CIImage(image: ...)` works also if it is created from UIImage not backed internally by CIImage. Unfortunately this extension is not documented by Apple, but it works also for CGImage backed UIImage.

Reply:
Oh, that is really cool then, TIL! Thanks guys 🍻